grpc/src/main/java/com/linecorp/armeria/internal/client/grpc/ArmeriaClientCall.java

```diff
@@ -199,21 +202,24 @@ public void start(Listener<O> responseListener, Metadata metadata) {
         requireNonNull(responseListener, "responseListener");
         requireNonNull(metadata, "metadata");
 
+        // The listener is set by the callers thread. The safe-publishing (aka visibility)
+        // to event-loop is guaranteed via semantics of EventLoop.execute(...) call.
+        listener = responseListener;
+
         final Compressor compressor;
         if (callOptions.getCompressor() != null) {
             compressor = compressorRegistry.lookupCompressor(callOptions.getCompressor());
             if (compressor == null) {
-                responseListener.onClose(
-                        Status.INTERNAL.withDescription(
-                                "Unable to find compressor by name " + callOptions.getCompressor()),
-                        new Metadata());
+                final Status status = Status.INTERNAL
+                        .withDescription("Unable to find compressor by name " + callOptions.getCompressor());
+                close(status, new Metadata());
```

Comment:
In case when compressor is not found, we now properly recycle underlying resources (req), just like we do in all other failure cases.